src/compiler/checker.ts

```diff
@@ -6783,6 +6783,37 @@ namespace ts {
                     const targetName = getInternalSymbolName(target, verbatimTargetName);
                     includePrivateSymbol(target); // the target may be within the same scope - attempt to serialize it first
                     switch (node.kind) {
+                        case SyntaxKind.BindingElement:
+                            if (node.parent?.parent?.kind === SyntaxKind.VariableDeclaration) {
+                                // const { SomeClass } = require('./lib');
+                                const specifier = getSpecifierForModuleSymbol(target.parent || target, context); // 'y'
+                                addResult(factory.createImportDeclaration(
+                                    /*decorators*/ undefined,
+                                    /*modifiers*/ undefined,
+                                    factory.createImportClause(/*isTypeOnly*/ false, /*name*/ undefined, factory.createNamedImports([factory.createImportSpecifier(
+                                        (node as BindingElement).propertyName && isIdentifier((node as BindingElement).propertyName!) ? factory.createIdentifier(idText((node as BindingElement).propertyName as Identifier)) : undefined,
+                                        factory.createIdentifier(localName)
+                                    )])),
+                                    factory.createStringLiteral(specifier)
+                                ), ModifierFlags.None);
+                                break;
+                            }
+                            // At present, the below case should be entirely unhit, as, generally speaking, the below case is *usually* bound
+                            // such that the `BinaryExpression` is the declaration rather than the specific, nested binding element
+                            // (because we don't seek to emit an alias in these forms yet). As such, the `BinaryExpression` switch case
+                            // will be what actually handles this form. _However_, in anticipation of binding the below with proper
+                            // alias symbols, I'm _pretty comfortable_ including the case here, even though it is not yet live.

```

Comment:
were you able to test this locally to see whether the code worked? 

(I'm not a fan of adding future-code because it tends to be 98% correct, but the first person to rely on it has no idea which code needs to be fixed.)

Reply:
Today we bind
```
module.exports = { SomeClass }
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ // this node as the `SomeClass` alias symbol's declaration
```
while in the future we probably
```
module.exports = { SomeClass }
                   ^^^^^^^^^ // should bind this node as the `SomeClass` alias symbol's declaration
```
because as it stands today, if I write
```
module.exports = { SomeClass, SomeClass2 }
```
both the `SomeClass` and `SomeClass2` symbols have the same declaration (the whole binary expression), which probably does bad things to some LS features (like rename?).

This small branch of code (the comment is bigger) will handle if/when we bind to the individual `BindingElement` instead. I figured adding it while I was here would be better, since we evidently forgot to handle serialization when we changed how aliases were bound once already. 😛

Reply:
Yep, fair enough =)